### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by BurningLoveMedia** [pojome/elementor#1922]
I would love to be able to show post title and possibly metadata permanently on the portfolio widget rather than just on the mouse rollover. Below the featured image would be even better.

**Comment by drawcard** [comment]
I would like this too, but I wonder if you then start overlapping too much with the existing post widget (which can accept custom post types as well). I agree though that it would be nice to have some more flexibility with the presentation for this - the meta filter function is awesome and I want to use it in more places than just Portfolio scenarios.

**Comment by quantissimo** [comment]
I'd like to see this design implemented as well

**Comment by HammerOz** [comment]
The hover to show title makes this widget unusable on mobile friendly (all) sites. I would like  the overlay and title to show by default on mobile/tablet . I would like the card to display the taxonomy or portfolio type as well. With much respect you guys need start thinking mobile first with your widgets. I am hitting a lot of walls on this front.

**Comment by chocobochick** [comment]
+1

**Comment by mrandt** [comment]
+1

I am particularly interested in displaying the title below or above for mobile devices. The current implementation results in bad UX, as the user will only see the title when tapping the image - and that will trigger navigation right away.

Other option would be to implement two-tap pattern for responsive mode - first tap to reveal title, second tap to navigate.

**Comment by Matherone** [comment]
+1.

Please add the possibility to display title and possibly other source data as a permanent text overlay on top of the portfolio thumbnails, **regardless of hover state**.

**Comment by rudeone** [comment]
yes please i need this as well. im currently looking for a plugin which does this.

also: i need this feature to work by manually adding items, not only posts or CPT

**Comment by nathanstrapazzon** [comment]
I'd like to have this feature too. For me, the widget is almost unusable without the title only in hover

**Comment by nathanstrapazzon** [comment]
I'd like to have this feature too. For me, the widget is almost unusable with the title **only** in hover.

It could be like the call to action widget, would be perfect than.

**Comment by lindagester** [comment]
+1

I'd like to add more metadata, not just the title. Also the option of displaying text not just on hover.

**Comment by alfredobr** [comment]
We are still waiting for this Elementor team. Or anyone has achieved a solution to this?

**Comment by drawcard** [comment]
What I managed to do was hack the overlay DIV, so that it always displayed at the bottom of the thumbnail image. In your custom CSS settings for the Portfolio widget add:

```css
selector .elementor-portfolio-item__overlay {
    opacity: 1;
    top: auto;
    padding: 10px;
}
```
Colours and styles are set as normal in the Overlay style settings. You can also play around with the CSS to meet your needs.

Note that this doesn't allow for anything further like descriptions or metadata, but it works if you just want to show the title and thumbnail image simultaneously. For that you may need to look into filter hooks to update what the title shows.

**Comment by jeffuetrecht** [comment]
Thanks @drawcard for the post! I'm using a variation of your CSS and am much happier now with the functionality of the portfolio. That said, I would love for Elementor to really solve this, with the ability to have a title always show, maybe have an option for content (think product portfolio, not just media), and much greater ability to format.

**Comment by drawcard** [comment]
I also just came across this idea in an unrelated post - expanding on the notion that you can use action hooks / filters to add in whatever content you want underneath a portfolio title. This might allow you to display categories, dates etc.

https://github.com/elementor/elementor/issues/8210#issuecomment-497313256

**Comment by simea-simea** [comment]
Awesome, thanks! However, it does not display on my phone... I see it correctly on Elementor but not on the phone.. What can I try?

**Comment by drawcard** [comment]
@simea-simea can you post screenshots?

**Comment by jeffuetrecht** [comment]
I just ran across the plugin WP Show Posts (by Tom Usborne, author of the awesome GeneratePress theme). At first glance, the plugin doesn't seem to have filtering by category, which I would like but can do without. For my site, it may become the alternative to Elementor's portfolio widget.

**Comment by simea-simea** [comment]
@drawcard I now used another widget. 

It simply showed Images. As default without the code. It worked on desktop though.

**Comment by drawcard** [comment]
@simea-simea was that when you were using my CSS trick, or without it? Reason I ask is I am using that CSS trick on other client sites and it seems to work OK on mobile. So I'm just a bit curious.

**Comment by simea-simea** [comment]
Yes with your css trick it did not work on mobile.

**Comment by drawcard** [comment]
:thinking: Weird. Thanks for letting me know!

**Comment by jeffuetrecht** [comment]
FWIW, the CSS worked for me on both mobile and desktop. Using the GeneratePress theme, with minimal mobile-specific CSS added.

**Comment by jeffalewis** [comment]
Yes please! +1 Did anyone figure out how to do this? Tried the css and worked but not really feeling the look of it. Would love for post title to just be below. Like with post widget, which I would use except has no filter bar option! So close!

**Comment by jhubsch** [comment]
I used @drawcard 's [solution](https://github.com/elementor/elementor/issues/1922#issuecomment-496793447) here, but I noticed that the title would be placed at the bottom of the container that held the image, and it would basically overlay it. This would cut off the image, even moreso if it went beyond just one line.  
  
I asked a colleague of mine to look into it a bit more, and he found the following steps fix this issue:  
1) Give the Portfolio widget itself a class (in this example, it's 'resources-portfolio').
2) Add this custom CSS:  

```
.resources-portfolio .elementor-post__thumbnail__link {
	padding-bottom: 0 !important;
    
}
.resources-portfolio .elementor-portfolio.elementor-has-item-ratio .elementor-post__thumbnail {
	position: relative;
}

.resources-portfolio .elementor-posts-container.elementor-has-item-ratio .elementor-post__thumbnail img {
	position: relative;
	top: 0;
	left: 0;
	transform: none;
}
.resources-portfolio .elementor-portfolio-item__overlay {
	position: relative;
}
.resources-portfolio .elementor-posts-container.elementor-has-item-ratio .elementor-post__thumbnail.elementor-fit-height img {
	max-width: 100%;
	height: auto;
}
```

The above CSS block, alongside @drawcard 's original suggestion, seems to work the way we wish was possible inherently within Elementor. Just sharing for anyone that might make use of it (like @jeffalewis )!

**Comment by agb2113** [comment]
Jhubsch's solution causes the consistent aspect ratio to get overridden.  Is there a workaround for this?  Can't believe elementor hasn't added a fix yet.

**Comment by jensolafkoch** [comment]
Thanks for all the groundwork! 

I settled for drawcard's solution for now but combined with a media query to check for the presence of hover capabilities of the device:

```
@media (hover:none) {
  .resources-portfolio .elementor-portfolio-item__overlay {
    opacity: 0.6 !important;
    top: 60% !important;
    padding: 10px !important;
  }
}
```

That way the portfolio automatically switches between the initial Elementor settings and the fine-tuned ones for other devices where there is no chance to get an overlay.

**Comment by kwalker1995** [comment]
I adapted @drawcard solution so that you can have the text under your image and it doesn't distort the aspect ratio. However be sure to change the font size of the title to a smaller size if your titles are going to be long, as if the title takes up too much space it will eventually go back over the image. You can change the title font size and the overlay background colour in the editor, if you want it to be transparent use #FFFFFF00 as the color.

`selector .elementor-portfolio-item__overlay {
    opacity: 1;
    top: auto;
    padding: 10px;
    position: absolute;
    bottom: -60px;
    
}

selector .elementor-portfolio-item__title {
   
}

selector .elementor-portfolio-item{
    overflow: visible;
    margin-bottom: 80px!important;
}

`

**Comment by kyleTyrone95** [comment]
I adapted @drawcard solution so that you can have the text under your image and it doesn't distort the aspect ratio. However be sure to change the font size of the title to a smaller size if your titles are going to be long, as if the title takes up too much space it will eventually go back over the image. You can change the title font size and the overlay background colour in the editor, if you want it to be transparent use #FFFFFF00 as the color.

`selector .elementor-portfolio-item__overlay {
opacity: 1;
top: auto;
padding: 10px;
position: absolute;
bottom: -60px;

}

selector .elementor-portfolio-item__title {

}

selector .elementor-portfolio-item{
overflow: visible;
margin-bottom: 80px!important;
}

`

**Comment by Helledev** [comment]
Hi 
This worked for me... leaving the title in always visible in the middle of the image and background darken on hover.

selector .elementor-portfolio-item__overlay {
    opacity: 0.8;
    top: 50;
    padding: 10px;
}

.elementor-34 .elementor-element.elementor-element-388f5ae a .elementor-portfolio-item__overlay:hover {
    background-color: rgb(0, 0, 0, 0.7);
   transition-duration: 1s;
  transition-property: background-color;
}

**Comment by kellyrenee814** [comment]
+1 Have option to show the titles beneath the image.

**Comment by thevanabonds** [comment]
@jhubsch 

Hi, I used your code and it worked well (Thanks!), but my thumbnails are different sizes. Is there anyway to add additional code so that they all show up the same size? I no nothing about CSS so any insight is greatly appreciated!

**Comment by davidamarek** [comment]
+1 for this.

**Comment by bryanfawcett** [comment]
Hi I copied text from another blog way back, and have customized to be very similar to what is here with some added css.
`selector .elementor-portfolio-item__overlay {
    opacity: 1;
    top: auto;
    padding: 10px;
    border-radius: 0px;
}

selector .elementor-portfolio-item {
    border-radius: 15px;   
}
selector .elementor-portfolio-item__img {
    overflow: hidden;
    -webkit-transform: scale(1.2);
    -moz-transform: scale(1.2);
    -ms-transform: scale(1.2);
    -o-transform: scale(1.2);
	transform: scale(1.2);
	-webkit-transition: .4s ease-in-out;
	-moz-transition: .4s ease-in-out;
	-o-transition: .4s ease-in-out;
	transition: .4s ease-in-out;
	border-radius: 15px;
}
selector .elementor-portfolio-item__img:hover {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    -o-transform: scale(1);
	transform: scale(1);
}`

**Comment by harut06012003** [comment]
Hi guys, I am facing an issue in portfolio page, can someone help? I would like to always show the title, in hover i want it to disapear, it works oppsite, can anyone help?
